CMakeLists.txt

```diff
@@ -0,0 +1,61 @@
+SET(CMAKE_OSX_DEPLOYMENT_TARGET "10.12" CACHE STRING "Minimum OS X deployment version")
+CMAKE_MINIMUM_REQUIRED(VERSION 3.1.0)
+
+# root
+PROJECT(perimeter)
+
+# modules
+SET(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake)
+
+#SET(CMAKE_CONFIGURATION_TYPES "Release;Debug;MinSizeRel;RelWithDebInfo")
+IF(MINGW)
+    SET(CMAKE_FIND_LIBRARY_SUFFIXES ".dll")
+ENDIF()
+
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+# depends
+FIND_PACKAGE(Boost)
+
+# globals
+INCLUDE_DIRECTORIES(SYSTEM
+        ${Boost_INCLUDE_DIRS}
+        )
+
+INCLUDE(CheckCXXCompilerFlag)
+SET(CMAKE_CXX_FLAGS_DEBUG "-g -Wall -Wextra")
```

Comment:
for sure will not work under MSVC, move under if

Reply:
Done, Im not sure if MSVC will need any specific stuff but IF is in place if thats the case